lib/postgres.js

```diff
@@ -47,14 +47,100 @@ class Postgres {
             stickers    jsonb,
             updated     timestamp NOT NULL,
             rarity      smallint NOT NULL,
+            floatid     bigint  NOT NULL,
             PRIMARY KEY (a)
-        )`);
+        );
+
+        CREATE TABLE IF NOT EXISTS history (
+            floatid     bigint  NOT NULL,
+            a           bigint  NOT NULL,
+            steamid     bigint  NOT NULL,
+            time        timestamp NOT NULL,
+            PRIMARY KEY (floatid, a)
+        );
+
+        ALTER TABLE items ADD COLUMN IF NOT EXISTS floatid BIGINT;
+
+        -- Float ID is defined as the first asset id we've seen for an item
+
+        CREATE OR REPLACE FUNCTION is_steamid(IN val bigint)
+            RETURNS boolean
+            LANGUAGE 'plpgsql'
+            PARALLEL SAFE
+        AS $BODY$BEGIN
+            IF val < 76561197960265728 THEN
+                RETURN FALSE;
+            ELSIF (val >> 56) > 5 THEN
+                RETURN FALSE;
+            ELSIF ((val >> 32) & ((1 << 20) - 1)) > 32 THEN
+                RETURN FALSE;
+            END IF;
+        
+            RETURN TRUE;
+        END;$BODY$;
+        
+        CREATE OR REPLACE FUNCTION update_history()
+        RETURNS TRIGGER
+        AS $$
+        DECLARE
+            floatid BIGINT;
+        BEGIN
+            -- Handle cases where the floatid isn't there
+            IF NEW.floatid IS NOT NULL THEN
+                floatid = NEW.floatid;
+            ELSE
+                -- Update the DB with the floatid if it still doesn't exist
+                UPDATE items SET floatid=OLD.a WHERE a=NEW.a;
+                floatid = OLD.a;
+            END IF;
+
+            IF is_steamid(OLD.ms) AND OLD.ms != NEW.ms THEN
```

Comment:
Wouldn't this also include items that have been listed to the market? Do we want to keep history of items listed to the market?

Reply:
Are you referring to the new item asset being from the market or the old one?

We only care if the existing record that is about to be updated contains a steam id inspect, if so, save it.

Reply:
So if a user lists the item onto the steam market the old.ms is the users id which is valid but the new.ms is a market listing. Wouldn't this then insert a new history value for the market listing?

Reply:
It'd insert a history value for the value it's overriding, which is the steam inventory history.